src/server/test_utils.cc

```diff
@@ -201,6 +211,20 @@ void BaseFamilyTest::ResetService() {
   LOG(INFO) << "Starting " << test_info->name();
 }
 
+void BaseFamilyTest::ShutdownService() {
+  DCHECK(service_);
+
+  service_->Shutdown();
+  service_.reset();
+  delete shard_set;
+  shard_set = nullptr;
+
+  watchdog_done_.Notify();
+  watchdog_fiber_.Join();
```

Comment:
you are calling join twice if ResetService is called from a test
once from ResetService and another time from TearDown

Reply:
fixed